### PR TITLE
added ssl support in api-server

### PR DESCRIPTION
**Pull request by kociolekscramjet**
Extension for api-server config which makes sslKeyPath and sslCertPath as possible option properties.

The server property has priority over sslKeyPath and sslCertPath. If server is not provided and sslKeyPath and sslCertPath exists then https server instance is created and passed as cero server config property.

### CODE REVIEW

File: packages/api-server/src/index.ts [hunk 1]
```diff
@@ -45,6 +47,25 @@ function safeDecorator(cb: (req: IncomingMessage) => MaybePromise<void>) {
...
+        const sslConfig = {
+            key: fs.readFileSync(conf.sslKeyPath),
+            cert: fs.readFileSync(conf.sslCertPath)
+        };
+
+        return https.createServer(sslConfig);
+    }
+
+    return undefined;
+}
```
Comment (patuwwy) [inline packages/api-server/src/index.ts:67]:
```suggestion
function createCeroServerConfig(conf: ServerConfig = {}): ServerConfig["server"] {
    if (conf.server) {
        return conf.server;
    }

    if (conf.sslKeyPath && conf.sslCertPath) {
        const sslConfig = {
            key: fs.readFileSync(conf.sslKeyPath),
            cert: fs.readFileSync(conf.sslCertPath)
        };

        return createHttpsServer(sslConfig);
    }

    return createHttpServer();
}
```

File: packages/api-server/src/index.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@ import { CeroRouter, CeroRouterConfig } from "./lib/definitions";
 export type ServerConfig = {
     verbose?: boolean;
     server?: Server;
```
Comment (patuwwy) [inline packages/api-server/src/index.ts:13]:
```suggestion
    server?: HttpsServer | HttpServer;
```

Comment (patuwwy) [reply packages/api-server/src/index.ts:13]: requires
```
import { IncomingMessage, ServerResponse, Server as HttpServer, createServer as createHttpServer } from "http";
import { Server as HttpsServer, createServer as createHttpsServer } from "https";
```

File: packages/api-server/src/index.ts (hunk 1 above)
Comment (patuwwy) [reply packages/api-server/src/index.ts:67]: what do you think?

Comment (kociolekscramjet) [reply packages/api-server/src/index.ts:67]: Brilliant! Didn't know how to approach that type thanks! :)

**Review by alicja-gruzdz — APPROVED** [review]
Nice we have a support for both! 2 in 1!

**Review by MichalCz — APPROVED** [review]

**Pull request merged**